pkg/controller/statusmanager/status_manager.go

```diff
@@ -90,6 +90,99 @@ func (status *StatusManager) deleteRelatedObjectsNotRendered(co *configv1.Cluste
 			objToDelete.SetName(currentObj.Name)
 			objToDelete.SetNamespace(currentObj.Namespace)
 			objToDelete.SetGroupVersionKind(gvk)
+			if objToDelete.GetKind() == "Namespace" {
+				continue
```

Comment:
Why was this added? We want to delete unused namespaces...
(Though... I guess probably we need to fix it to do that *last*.)

Reply:
Due to some reason, the 'namespace' deletion is always stuck at the "Terminating" state forever, even when all the related objects in that namespace have been deleted. And it will block the same Namespace creation if we what to perform a rollback from ovn-kubernetes to openshift-sdn. 

I agree that we need to remove the unused namespaces anyway. But I think maybe we can fix it later.

Reply:
> Due to some reason, the 'namespace' deletion is always stuck at the "Terminating" state forever, even when all the related objects in that namespace have been deleted.

Maybe there is something else in that Namespace that isn't in the related objects then?

Reply:
Yes, there are some objects which are not managed by the CNO. I am not sure what is the proper way to delete all of them, then delete the namespace itself.

Reply:
hm. do you remember what objects?

Reply:
I did some further tests. It turns out that the namespace will be stuck in 'Terminating' before the cluster reboot, but can be deleted eventually after that. Therefore I remove the lines above.